portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,66 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// use modern JavaScript (ES5)
+"use strict"
+
 /**
- * Adds a random greeting to the page.
+ * Adds a cycling greeting to the page.
  */
+let greetingIndex = 0;
 function addRandomGreeting() {
   const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!', 'Hallo Welt!'];
+
+  // Pick the next greeting in a cycle.
+  const greeting = greetings[greetingIndex];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  // Update greeting index.
+  greetingIndex = (greetingIndex + 1) % greetings.length;
 
   // Add it to the page.
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/**
+ * Presents a receipt for getting user form feedback in pop-up window.
+ */
+function presentFeedbackReceipt() {
+  // Obtain user input feedback content, name and email.
+  const formElements = document.getElementById('feedback-form').elements;
+  const userFeedback = formElements[0].value;
+  const userName = formElements[1].value;
+  const userEmail = formElements[2].value;
+
+  // Construct feedback receipt.
+  const receipt = `Dear ${userName},\nThank you for submitting a feedback!\n`
+                  + `We have recorded the following:\n`
+                  + `    *Message: "${userFeedback}"\n`
+                  + `    *Contact Information: ${userEmail}\n`;
+
+  // Present feedback receipt in pop-up window.
+  window.alert(receipt);
+}
+
+/**
+ * Executes JavaScript code and returns execution results like a console.
+ */
+function executeJavaScript() {
+  // Obtain user input JavaScript code.
+  const promptString = "Please enter your JavaScript code:\n";
+  const jsCode = window.prompt(promptString);
+  if (!jsCode) {
+    window.alert("No code received.");
+    return;
+  }
+
+  // Executes JavaScript code.
+  const executionResult = eval(jsCode);
```

Comment:
minor nit: Would it be possible to display an error message if the user inputs invalid/unrunnable js code? I think right now it just doesn't display anything (i.e. no console output)

Reply:
Got it! I'll think about how to handle that